tests/unit/h/conftest.py

```diff
@@ -129,7 +129,7 @@ def fake_feature():
 @pytest.fixture
 def form_validating_to():
     def form_validating_to(appstruct):
-        form = mock.MagicMock()
+        form = mock.NonCallableMagicMock()
```

Comment:
Form objects are not callable so `NonCallableMagicMock()` is appropriate here rather than `MagicMock()`. (`handle_form_submission()` now handles the `form` argument differently if it's a callable so it was necessary to fix this.)

`NonCallableMagicMock()` is used rather than `NonCallableMock()` because the code needs subscripting (`[...]`) to work on the mock form object.

(This whole `form_validating_to()` thing and the use of mocks in this way is bad test writing but not fixing that here.)